background/background.go

```diff
@@ -324,37 +320,45 @@ func Start(c context.Context, wg *sync.WaitGroup, vdb *database.VspDatabase, drp
 	checkWalletConsistency()
 
 	// Run voting wallet consistency check periodically.
+	wg.Add(1)
 	go func() {
 		ticker := time.NewTicker(consistencyInterval)
+	consistencyLoop:
 		for {
 			select {
-			case <-ctx.Done():
+			case <-shutdownCtx.Done():
 				ticker.Stop()
-				return
+				break consistencyLoop
 			case <-ticker.C:
 				checkWalletConsistency()
```

Comment:
could pass shutdownCtx to checkWalletConsistency()

Reply:
I don't think that would work well because the shutdown context is cancelled as soon as CTRL+c (or another shutdown signal) is received, and wsrpc clients stop working when their context is cancelled.

If a shutdown signal is received while `checkWalletConsistency()` is running, RPC clients will throw errors and the consistency checking process will error out before completing.